frontend/app/src/components/settings/OnboardingSettings.vue

```diff
@@ -38,74 +30,78 @@ const configuration: Ref<BackendConfiguration> = asyncComputed(() =>
   backendSettings()
 );
 
-const { info } = useInfoApi();
-const defaultArguments: Ref<DefaultBackendArguments> = asyncComputed(
-  () => info().then(info => info.backendDefaultArguments),
-  {
-    maxLogfilesNum: 0,
-    maxSizeInMbAllLogs: 0,
-    sqliteInstructions: 0
-  }
-);
-
-const { tc } = useI18n();
+const { t } = useI18n();
 
-const loaded = async () => {
+const initialOptions: ComputedRef<Partial<BackendOptions>> = computed(() => {
   const config = get(configuration);
   const opts = get(options);
-  const defaults = get(defaultArguments);
-
-  set(userDataDirectory, opts.dataDirectory ?? get(dataDirectory));
-  set(userLogDirectory, opts.logDirectory ?? get(defaultLogDirectory));
-  set(logLevel, opts.loglevel ?? get(defaultLogLevel));
-  set(logFromOtherModules, opts.logFromOtherModules ?? false);
-  set(
-    maxLogFiles,
-    opts.maxLogfilesNum ??
-      config.maxLogfilesNum.value ??
-      defaults.maxLogfilesNum
-  );
-  set(
-    maxLogSize,
-    opts.maxSizeInMbAllLogs ??
-      config.maxSizeInMbAllLogs.value ??
-      defaults.maxSizeInMbAllLogs
-  );
-  set(
-    sqliteInstructions,
-    opts.sqliteInstructions ??
-      config.sqliteInstructions.value ??
+  const defaults = get(defaultBackendArguments);
+  return {
+    loglevel: opts.loglevel ?? get(defaultLogLevel),
+    dataDirectory: opts.dataDirectory ?? get(dataDirectory),
+    logDirectory: opts.logDirectory ?? get(defaultLogDirectory),
+    logFromOtherModules: opts.logFromOtherModules ?? false,
+    maxLogfilesNum:
+      opts.maxLogfilesNum ??
+      config?.maxLogfilesNum?.value ??
+      defaults.maxLogfilesNum,
+    maxSizeInMbAllLogs:
+      opts.maxSizeInMbAllLogs ??
+      config?.maxSizeInMbAllLogs?.value ??
+      defaults.maxSizeInMbAllLogs,
+    sqliteInstructions:
+      opts.sqliteInstructions ??
+      config?.sqliteInstructions?.value ??
       defaults.sqliteInstructions
-  );
+  };
+});
+
+const loaded = async () => {
+  const initial = get(initialOptions);
+
+  set(logLevel, initial.loglevel);
+  set(userDataDirectory, initial.dataDirectory);
+  set(userLogDirectory, initial.logDirectory);
+  set(logFromOtherModules, initial.logFromOtherModules);
+  set(maxLogFiles, initial.maxLogfilesNum);
+  set(maxLogSize, initial.maxSizeInMbAllLogs);
+  set(sqliteInstructions, initial.sqliteInstructions);
 };
 
 const {
   resetOptions,
   saveOptions,
   fileConfig,
-  logLevel,
   defaultLogLevel,
   defaultLogDirectory,
   options
 } = useBackendManagement(loaded);
 
+const logLevel: Ref<LogLevel> = ref<LogLevel>(get(defaultLogLevel));
+const userDataDirectory: Ref<string> = ref('');
+const userLogDirectory: Ref<string> = ref('');
+const logFromOtherModules: Ref<boolean> = ref(false);
+const maxLogSize: Ref<number> = ref(0);
+const sqliteInstructions: Ref<number> = ref(0);
+const maxLogFiles: Ref<number> = ref(0);
```

Comment:
Is there any reason why we can't leave reference declarations at the top of the file?